packages/rainbowkit/src/wallets/walletConnectors/oktoWallet/oktoWallet.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { getWalletConnectUri } from '../../../utils/getWalletConnectUri';
+import { isAndroid } from '../../../utils/isMobile';
+import { Wallet } from '../../Wallet';
+import { getWalletConnectConnector } from '../../getWalletConnectConnector';
+import type {
+  WalletConnectConnectorOptions,
+  WalletConnectLegacyConnectorOptions,
+} from '../../getWalletConnectConnector';
+
+export interface OktoWalletLegacyOptions {
+  projectId?: string;
+  chains: Chain[];
+  walletConnectVersion: '1';
+  walletConnectOptions?: WalletConnectLegacyConnectorOptions;
+}
+
+export interface OktoWalletOptions {
+  projectId: string;
+  chains: Chain[];
+  walletConnectVersion?: '2';
+  walletConnectOptions?: WalletConnectConnectorOptions;
+}
+
+export const oktoWallet = ({
+  chains,
+  projectId,
+  walletConnectOptions,
+  walletConnectVersion = '2',
+}: OktoWalletLegacyOptions | OktoWalletOptions): Wallet => ({
+  id: 'Okto',
+  name: 'Okto',
+  iconUrl: async () => (await import('./oktoWallet.svg')).default,
+  iconBackground: '#fff',
+  downloadUrls: {
+    android:
+      'https://play.google.com/store/apps/details?id=im.okto.contractwalletclient',
+    ios: 'https://apps.apple.com/us/app/okto/id1358741926',
+    mobile: 'https://okto.tech/',
+    qrCode: 'https://okto.tech/',
+  },
+  createConnector: () => {
+    const connector = getWalletConnectConnector({
+      projectId,
+      chains,
+      version: walletConnectVersion,
+      options: walletConnectOptions,
+    });
+
+    return {
+      connector,
+      mobile: {
+        getUri: async () => {
+          const uri = await getWalletConnectUri(
+            connector,
+            walletConnectVersion
+          );
+          return isAndroid() ? uri : `okto://wc?uri=${encodeURIComponent(uri)}`;
+        },
+      },
+      qrCode: {
+        getUri: async () =>
+          getWalletConnectUri(connector, walletConnectVersion),
```

Comment:
Missing an await; please test your connector with the example deployment

Reply:
Hi @DanielSinclair thanks for the review. The code works in the example deployment. There exists similar implementation for `getUri` without `await` in [argentWallet](https://github.com/rainbow-me/rainbowkit/blob/2d9c1c14c88e947116ea726f5f3138c6ced95631/packages/rainbowkit/src/wallets/walletConnectors/argentWallet/argentWallet.ts#L65) and [okxWallet](https://github.com/rainbow-me/rainbowkit/blob/2d9c1c14c88e947116ea726f5f3138c6ced95631/packages/rainbowkit/src/wallets/walletConnectors/okxWallet/okxWallet.ts#L97).

Reply:
@oktotech Please use a square SVG

Reply:
@DanielSinclair is it okay if we give a white square SVG with okto logo in centre. Or its Rainbow-Kit guidelines to provide a gradient filled square logo?

Reply:
@DanielSinclair We have updated the icon according to guidelines